paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/mapred/PaimonInputFormat.java

```diff
@@ -48,8 +49,9 @@ public InputSplit[] getSplits(JobConf jobConf, int numSplits) {
         FileStoreTable table = createFileStoreTable(jobConf);
         InnerTableScan scan = table.newScan();
         createPredicate(table.schema(), jobConf).ifPresent(scan::withFilter);
+        String location = jobConf.get(hive_metastoreConstants.META_TABLE_LOCATION);
         return scan.plan().splits().stream()
-                .map(split -> new PaimonInputSplit(table.location().toString(), (DataSplit) split))
+                .map(split -> new PaimonInputSplit(location, (DataSplit) split))
```

Comment:
add comments here, why we need to use `hive_metastoreConstants.META_TABLE_LOCATION`.